packages/react/src/components/Facepile/Facepile.base.tsx

```diff
@@ -104,16 +104,25 @@ export class FacepileBase extends React.Component<IFacepileProps, {}> {
     singlePersona: boolean,
     showTooltip: boolean,
   ): JSX.Element[] {
-    const { onRenderPersona = this._getPersonaControl, onRenderPersonaCoin = this._getPersonaCoinControl } = this.props;
+    const {
+      onRenderPersona = this._getPersonaControl,
+      onRenderPersonaCoin = this._getPersonaCoinControl,
+      onRenderPersonaWrapper,
+    } = this.props;
     return personas.map((persona: IFacepilePersona, index: number) => {
       const personaControl: JSX.Element | null = singlePersona
         ? onRenderPersona(persona, this._getPersonaControl)
         : onRenderPersonaCoin(persona, this._getPersonaCoinControl);
       return (
         <li key={`${singlePersona ? 'persona' : 'personaCoin'}-${index}`} className={this._classNames.member}>
           {persona.onClick
-            ? this._getElementWithOnClickEvent(personaControl, persona, showTooltip, index)
-            : this._getElementWithoutOnClickEvent(personaControl, persona, showTooltip, index)}
+            ? onRenderPersonaWrapper
+              ? onRenderPersonaWrapper(
+                  { personaControl, persona, showTooltip, index },
+                  this._getElementWithOnClickEvent.bind(this),
+                )
+              : this._getElementWithOnClickEvent({ personaControl, persona, showTooltip, index })
+            : this._getElementWithoutOnClickEvent({ personaControl, persona, showTooltip, index })}
```

Comment:
Rather than passing all the `{ personaControl, persona, showTooltip, index }` props into `onRenderPersonaWrapper`, I wonder if we should do something like this:

```jsx
const personaControl: JSX.Element | null = singlePersona
        ? onRenderPersona(persona, this._getPersonaControl)
        : onRenderPersonaCoin(persona, this._getPersonaCoinControl);
const defaultPersonaRender = persona.onClick
        ? () => this._getElementWithOnClickEvent(personaControl, persona, showTooltip, index)
        : () => this._getElementWithoutOnClickEvent(personaControl, persona, showTooltip, index)}

return (
  [... etc etc other code ...]
  onRenderPersonaWrapper ? onRenderPersonaWrapper(this.props, defaultPersonaRender) : defaultPersonaRender()
);
```

Reply:
This is definitely cleaner! I made the change following this pattern, only difference is that I'm only passing `persona` to the `onRenderPersonaWrapper` instead of the entire props object to keep it consistent with other `onRender` callbacks 😄